File: 844-backspace-string-compare/844-backspace-string-compare.rs
impl Solution {
    pub fn backspace_compare(s: String, t: String) -> bool {
        
        fn edit(acc: &mut Vec<char>, c: char) -> &mut Vec<char>{
            if c == '#'{
                acc.pop();
            }else{
                acc.push(c);
            }
            
            acc
        };
        
        let mut a = vec![];
        let mut b = vec![];
        
        s.chars().fold(&mut a, edit);
        t.chars().fold(&mut b, edit);
        
        
        a == b
    }
}

File: 844-backspace-string-compare/844-backspace-string-compare.py
class Solution:
    def backspaceCompare(self, s: str, t: str) -> bool:
        a = []
        b = []
        
        skip = 0
        for c in s[::-1]:
            if c == '#':
                skip +=1
            elif skip > 0:
                skip -=1
            else:
                a.append(c)
                
        skip = 0
                
        for c in t[::-1]:
            if c == '#':
                skip +=1
            elif skip > 0:
                skip -=1
            else:
                b.append(c)
        
        return a == b
                
            
            
                